.github/workflows/build.yaml

```diff
@@ -280,3 +281,54 @@ jobs:
       with:
         name: Widelands ${{ matrix.config }} ${{ matrix.compiler }} MacOS 11 AppImage
         path: ${{ github.workspace }}/Widelands-${{ github.sha }}-${{ matrix.config }}-${{ matrix.compiler }}.dmg
+
+  dev_release:
+    name: Automated development pre-release
+    # TODO: switch to 'refs/heads/master' when merging
+    if: ${{ github.ref == 'refs/heads/ci_dev_release' && always() }}
+    needs: [windows, macos]
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Check origin
+        # TODO: switch to 'widelands' when merging
+        run: |
+          if [ "${GITHUB_REPOSITORY%/*}" != "matthiakl" ]
+          then
+            echo "Skipping release creation on fork"
+            exit 1
+          fi
+      - name: Download artifacts
+        uses: actions/download-artifact@v2
+      - name: Check missing artifacts
+        # This step ensures, that the development release always contain an artifact for every platform
+        # artifacts.list lists all attachements of the pre-release. In case a build step failed, we detect
+        # that a file is missing and download the last successful build from the pre-release.
+        run: |
+          RELEASE_URL="$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/releases/download/latest"
+          mkdir Widelands-latest
+          cd Widelands-latest
+          curl -L -O "$RELEASE_URL/artifacts.list"
+          for ARTIFACT in $(cat artifacts.list)
+          do
+            [ "${ARTIFACT%%-*}" != "Widelands" ] && continue
+            MATRIX=$(echo $ARTIFACT | sed 's/Widelands-[0-9a-f]*-//' -)
+            ls ../Widelands*/Widelands*${MATRIX} || \
+              curl -L -O "$RELEASE_URL/$ARTIFACT"
+          done
+      - name: List artifacts
+        run: |
+          find . -name 'Widelands*' -type f -printf '%f\n' | tee artifacts.list
+      - name: Updating latest pre-release
```

Comment:
Directly before this step, you need to check that the commit that triggered this workflow is still the latest one, and `exit 1` if this is not the case. Builds do not always complete in the order in which they were triggered, so this could end up overwriting a newer release by accident without such a check. See the launchpad mirror or format actions how to check this

Reply:
Added from format.yaml